### PR TITLE
refactor get item and create an item

### PR DESCRIPTION
**Pull request by JoanneWooW**
* refactor get item
* refactor create an item
* fix a small typo bug
  * modified 'id' into 'item_id'
  * this bug will disorder the key name in response body
  * which will cause key error in TDD tests

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -309,12 +310,22 @@ class ItemResource(Resource):
...
-        Retrieve a single Order
-        This endpoint will return an Order based on it's id
+        Retrieve a single item in an order
+        This endpoint will return an item based on its id and its order's id
         """
-        pass
+        app.logger.info(f"Request for item with id {item_id} in order {order_id}")
+        order = CustomerOrder.find(order_id)
+        if not order: 
+            raise NotFound(f"Order with id {order_id} was not found")
```
Comment (pkutlesa) [inline service/routes.py:321]: abort(status.HTTP_404_NOT_FOUND, "Request for item with id {item_id} in order {order_id}")

File: service/routes.py [hunk 2]
```diff
@@ -309,12 +310,22 @@ class ItemResource(Resource):
...
+    
+        item = Item.find(item_id)
+        if not item: 
+            raise NotFound(f"Item with id {item_id} was not found in order {order_id}")
```
Comment (pkutlesa) [inline service/routes.py:325]: abort(status.HTTP_404_NOT_FOUND, "Item with id {item_id} was not found in order {order_id}")

**Review by pkutlesa — CHANGES_REQUESTED** [review]
Nice! Just left small comments on lines 321 and 325

**Comment by JoanneWooW** [comment]
@pkutlesa Nice catch, thank you! Fixed!

**Comment by pkutlesa** [comment]
close! just remove the letter f before the quotations on lines 321 and 325

**Comment by JoanneWooW** [comment]
@pkutlesa Thanks for pointing out! Could you explain why I should not use 'f' in front of quotations? I thought you missed them by accident so I added them on...

**Comment by pkutlesa** [comment]
My first impression was that they were causing tests to fail until I read your most recent update. f-strings should be fine :)

**Review by pkutlesa — APPROVED** [review]
LGTM!

**Comment by pkutlesa** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/9bf05db0-f920-4eb7-a171-e66653f00e58?env_id=ibm:yp:us-south "nyu-order-service-sum21-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/907055a0-7191-42af-9b3d-3504d7e9adb4 "nyu-order-service-sum21") deployed [orders](https://github.com/DevOps-Agile-Squad/orders/releases/tag/deploy-ibm-yp-us-south-zw1400-nyu.edu-prod-20210810-061036 "deploy-ibm-yp-us-south-zw1400-nyu.edu-prod-20210810-061036") to [prod](https://console.bluemix.net/apps/f43af9cd-c881-41ff-b5bd-1ee109712701?env_id=ibm:yp:us-south "ibm:yp:us-south:zw1400@nyu.edu:prod"), including commit b139f576276b790d30c2d4f0a9c3c71297197e79

**Pull request merged**